main/src/library/Concurrent/ReentrantLock.flix

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2021 Jonathan Lindegaard Starup
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Concurrent {
+
+    ///
+    /// A wrapper around a Java ReentrantLock.
+    ///
+    opaque type ReentrantLock = ##java.util.concurrent.locks.ReentrantLock
```

Comment:
This should probably also be public and placed inside the ReentrantLock namespace